script.audiobooks/plugin.py

```diff
@@ -1,418 +1,428 @@
-
-        if (filename is not None) and (len(filename) > 0):
-            menuNav = MenuNavigator(base_url, addon_handle)
-            menuNav.clear(filename[0])
-            del menuNav
+# -*- coding: utf-8 -*-
+import sys
+import os
+import urllib
+import urlparse
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import xbmcvfs
+
+__addon__ = xbmcaddon.Addon(id='script.audiobooks')
+__addonid__ = __addon__.getAddonInfo('id')
+__fanart__ = __addon__.getAddonInfo('fanart')
+__cwd__ = __addon__.getAddonInfo('path').decode("utf-8")
+__profile__ = xbmc.translatePath(__addon__.getAddonInfo('profile')).decode("utf-8")
+__resource__ = xbmc.translatePath(os.path.join(__cwd__, 'resources').encode("utf-8")).decode("utf-8")
+__lib__ = xbmc.translatePath(os.path.join(__resource__, 'lib').encode("utf-8")).decode("utf-8")
+
+
+sys.path.append(__resource__)
+sys.path.append(__lib__)
+
+# Import the common settings
+from settings import Settings
+from settings import log
+from settings import os_path_join
+
+from audiobook import AudioBookHandler
+from bookplayer import BookPlayer
+from database import AudioBooksDB
+
+
+###################################################################
+# Class to handle the navigation information for the plugin
+###################################################################
+class MenuNavigator():
+    def __init__(self, base_url, addon_handle):
+        self.base_url = base_url
+        self.addon_handle = addon_handle
+
+        self.tmpdestination = Settings.getTempLocation()
+        self.coverCache = Settings.getCoverCacheLocation()
+
+    # Creates a URL for a directory
+    def _build_url(self, query):
+        return self.base_url + '?' + urllib.urlencode(query)
+
+    # Show all the EBooks that are in the eBook directory
+    def showAudiobooks(self, directory=None):
+        # Get the setting for the audio book directory
+        audioBookFolder = Settings.getAudioBookFolder()
+
+        if audioBookFolder in [None, ""]:
+            # Prompt the user to set the eBooks Folder
+            audioBookFolder = xbmcgui.Dialog().browseSingle(0, __addon__.getLocalizedString(32005), 'files')
+
+            # Check to make sure the directory is set now
+            if audioBookFolder in [None, ""]:
+                xbmcgui.Dialog().ok(__addon__.getLocalizedString(32001), __addon__.getLocalizedString(32006))
+                return
+
+            # Save the directory in settings for future use
+            log("AudioBooksPlugin: Setting Audio Books folder to %s" % audioBookFolder)
+            Settings.setAudioBookFolder(audioBookFolder)
+
+        # We may be looking at a subdirectory
+        if directory not in [None, ""]:
+            audioBookFolder = directory
+
+        dirs, files = xbmcvfs.listdir(audioBookFolder)
+        files.sort()
+        dirs.sort()
+
+        bookDirs = []
+        # For each directory list allow the user to navigate into it
+        for dir in dirs:

```

Comment:
dir is a builtin function, it´s recommended to not use those for variable names.